samples/display-device-location-with-fused-location-data-source/src/main/java/com/esri/arcgismaps/sample/displaydevicelocationwithfusedlocationdatasource/screens/DisplayDeviceLocationWithFusedLocationDataSourceScreen.kt

```diff
@@ -0,0 +1,199 @@
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Settings
+import androidx.compose.material3.DropdownMenuItem
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.ExposedDropdownMenuBox
+import androidx.compose.material3.ExposedDropdownMenuDefaults
+import androidx.compose.material3.FloatingActionButton
+import androidx.compose.material3.HorizontalDivider
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.MenuAnchorType
+import androidx.compose.material3.ModalBottomSheet
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Slider
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextField
+import androidx.compose.material3.rememberModalBottomSheetState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableFloatStateOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Alignment.Companion.CenterVertically
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.arcgismaps.toolkit.geoviewcompose.MapView
+import com.arcgismaps.toolkit.geoviewcompose.rememberLocationDisplay
+import com.esri.arcgismaps.sample.displaydevicelocationwithfusedlocationdatasource.components.DisplayDeviceLocationWithFusedLocationDataSourceViewModel
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialog
+import com.esri.arcgismaps.sample.sampleslib.components.SampleTopAppBar
+import com.google.android.gms.location.Priority
+import kotlin.math.roundToInt
+
+/**
+ * Main screen layout for the sample app
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun DisplayDeviceLocationWithFusedLocationDataSourceScreen(sampleName: String) {
+    val locationDisplay = rememberLocationDisplay()
+    val mapViewModel: DisplayDeviceLocationWithFusedLocationDataSourceViewModel = viewModel()
+    // On first composition, initialize the sample.
+    var isViewmodelInitialized by remember { mutableStateOf(false) }
+    LaunchedEffect(isViewmodelInitialized) {
+        if (!isViewmodelInitialized) {
+            mapViewModel.initialize(locationDisplay)
+            isViewmodelInitialized = true
+        }
+    }
+    // Set up the bottom sheet controls
+    val sheetState = rememberModalBottomSheetState(skipPartiallyExpanded = true)
+    var showBottomSheet by remember { mutableStateOf(false) }
+    var isExpanded by remember { mutableStateOf(false) }
+    var priority by remember { mutableStateOf("High accuracy") }
+    var interval by remember { mutableFloatStateOf(1F) }
```

Comment:
These to me are more viewmodel properties than UI control states. Maybe it should live in the view model.

Reply:
I think they're ok in the composable since they represent the current state of the UI selection.